kernel/userspace.c

```diff
@@ -317,23 +315,27 @@ void _k_object_wordlist_foreach(_wordlist_cb_func_t func, void *context)
 }
 #endif /* CONFIG_DYNAMIC_OBJECTS */
 
-static int thread_index_get(struct k_thread *t)
+static bool thread_index_get(struct k_thread *thread, u32_t *thread_index)
```

Comment:
what rule is this?
We have lots of APIs which return negative on failure, why was this one specifically changed?

Reply:
yep, I wouldn't change it now too. See the comment bellow.

Reply:
In this API we were returing 2 different types of values. For a failure it was a -1( s32_t) and if it was success it will return the thread index which is a u32_t. 
This was caught by MISRA becuase we are typecasting a u32_t to a s32_t. 
Hence i re-wrote this function completely. now the status is checked on a boolean and the value will be returned by a pointer.
This change is actually needed, i woudn't want to differ untill later.